source/common/config/well_known_names.cc

```diff
@@ -218,8 +219,15 @@ TagNameValues::TagNameValues() {
   // http.<stat_prefix>.rbac.(<rules_stat_prefix>.)* but excluding policy
   addRe2(RBAC_HTTP_PREFIX, R"(^http\.<TAG_VALUE>\.rbac\.((<TAG_VALUE>)\.).*?)", "", "policy");
 
-  // proxy_proto.(versions.v<version_number>.)**
-  addRe2(PROXY_PROTOCOL_VERSION, R"(^proxy_proto\.(versions\.v(\d)\.)\w+)", "proxy_proto.versions");
+  // proxy_proto.(<stat_prefix>.)**
+  addRe2(PROXY_PROTOCOL_PREFIX, R"(^proxy_proto\.((<TAG_VALUE>)\.).+$)", "", "versions");
```

Comment:
I think this one can be done with `addTokenized` which @jmarantz says is much faster

Reply:
It gets kinda complicated because the `stat_prefix` is optional (backwards-compatibility). This means we need a matcher that correctly differentiates:

```
proxy_proto.test_stat_prefix.not_found_allowed        # Match
proxy_proto.test_stat_prefix.versions.v1.not_found  # Match
proxy_proto.versions.v1.not_found                              # DO NOT match
```

I specifically used `addRe2` because it has the `negative_matching_value` parameter, so I can ensure it doesn't match `proxy_proto.versions.v1.not_found` and accidently take `versions` as the `stat_prefix`.

I didn't see any similar functionality for `addTokenized`.

Reply:
Got it; that makes sense. Thanks for explaining!

Reply:
I want to point out for the future that you can give two addTokenized calls and that will probably be faster than one re2. However the fact that your regex starts with "^proxy_proto." means that it will benefit from an acceleration heuristic built into the tag extraction system, so I think this is fine. The regex will only be evaluated on stats that start with "proxy_proto.".